File: mobile/ios/BleManager/SpeechRequiredDataType.swift
//
//  SpeechRequiredDataType.swift
//  AugmentOS
//
//  Created by Yash Agarwal on 7/23/25.
//

import Foundation

/**
 * Enum representing the types of data required for speech processing
 * Matches the Android SpeechRequiredDataType enum
 */
enum SpeechRequiredDataType: String, CaseIterable {
    case PCM = "pcm"
    case TRANSCRIPTION = "transcription"
    case PCM_OR_TRANSCRIPTION = "pcm_or_transcription"

    /**
     * Convert from string value to enum
     * @param value The string value to convert
     * @return The corresponding enum value, or nil if not found
     */
    static func fromString(_ value: String) -> SpeechRequiredDataType? {
        return SpeechRequiredDataType(rawValue: value)
    }

    /**
     * Convert enum to string value
     * @return The string representation of the enum
     */
    func toString() -> String {
        return rawValue
    }

    /**
     * Convert array of strings to array of enums
     * @param stringArray Array of string values
     * @return Array of enum values, filtering out invalid strings
     */
    static func fromStringArray(_ stringArray: [String]) -> [SpeechRequiredDataType] {
        return stringArray.compactMap { fromString($0) }
    }

    /**
     * Convert array of enums to array of strings
     * @param enumArray Array of enum values
     * @return Array of string values
     */
    static func toStringArray(_ enumArray: [SpeechRequiredDataType]) -> [String] {
        return enumArray.map { $0.toString() }
    }
}

File: android_core/app/src/main/java/com/augmentos/augmentos_core/enums/SpeechRequiredDataType.java
package com.augmentos.augmentos_core.enums;

/**
 * Enum representing the types of data required for microphone state changes
 */
public enum SpeechRequiredDataType {
    PCM("pcm"),
    TRANSCRIPTION("transcription"),
    PCM_OR_TRANSCRIPTION("pcm_or_transcription");

    private final String stringValue;

    SpeechRequiredDataType(String stringValue) {
        this.stringValue = stringValue;
    }

    public String getStringValue() {
        return stringValue;
    }

    /**
     * Converts a string value to the corresponding enum value
     * @param value The string value to convert
     * @return The corresponding SpeechRequiredDataType enum value
     * @throws IllegalArgumentException if the string value is not recognized
     */
    public static SpeechRequiredDataType fromString(String value) {
        if (value == null) {
            throw new IllegalArgumentException("SpeechRequiredDataType string value cannot be null");
        }
        
        for (SpeechRequiredDataType type : SpeechRequiredDataType.values()) {
            if (type.stringValue.equals(value)) {
                return type;
            }
        }
        
        throw new IllegalArgumentException("Unknown SpeechRequiredDataType: " + value);
    }

    @Override
    public String toString() {
        return stringValue;
    }
}
